app/src/main/java/com/example/mindspace/ui/screens/Register.kt

```
@@ -0,0 +1,132 @@
+        try {
+            val account: GoogleSignInAccount = task.getResult(ApiException::class.java)
+            val credential = GoogleAuthProvider.getCredential(account.idToken, null)
+            auth.signInWithCredential(credential).addOnCompleteListener { authResult ->
+                if (authResult.isSuccessful) {
+                    Toast.makeText(context, "Google Sign Up Success!", Toast.LENGTH_SHORT).show()
+                    navController.navigate("home") { popUpTo("register") { inclusive = true } }
+                } else {
+                    Toast.makeText(context, "Google Sign Up Failed!", Toast.LENGTH_SHORT).show()
+                }
+            }
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)